holmes/core/tool_calling_llm.py

```
@@ -635,29 +598,83 @@ def _invoke_llm_tool_call(
+            error=tool_call_result.result.error,
+            metadata={
+                "status": tool_call_result.result.status,
+                "description": tool_call_result.description,
+            },
+        )
+
+    def _invoke_llm_tool_call(
+        self,
+        tool_to_call: ChatCompletionMessageToolCall,
+        previous_tool_calls: list[dict],
+        trace_span=DummySpan(),
```

Ruff (0.12.2)

Do not perform function call `DummySpan` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)